resources/prosody-plugins/mod_fmuc.lua

```diff
@@ -128,5 +134,105 @@ module:hook('muc-broadcast-presence', function (event)
     local full_p = st.clone(base_presence):add_child(full_x);
     full_p.attr.to = focus_occupant.jid;
     room:route_to_occupant(focus_occupant, full_p);
+
+    local raiseHand = full_p:get_child_text('jitsi_participant_raisedHand');
+    -- a promotion detected let's send it to main prosody
+    if raiseHand then
+        -- TODO check room= with tenants
+        local iq_id = new_id();
+        sent_iq_cache:set(iq_id, socket.gettime());
+        local promotion_request = st.iq({
+            type = 'set',
+            to = 'visitors.'..fmuc_main_domain,
+            from = main_domain,
+            id = iq_id })
+          :tag('visitors', { xmlns = 'jitsi:visitors',
+                             room = jid.join(jid.node(room.jid), muc_domain_prefix..'.'..fmuc_main_domain) })
+          :tag('request-promotion', { xmlns = 'jitsi:visitors', jid = occupant.jid }):up();
+        -- TODO what about name ???? it will be coming from the token, but we need to extract it and send it to the moderators
+        module:send(promotion_request);
+    end
+
     return;
 end);
+
+-- listens for responses to the iq sent for requesting promotion and forward it to the visitor
+local function stanza_handler(event)
+    local origin, stanza = event.origin, event.stanza;
+
+    if stanza.name ~= 'iq' then
+        return;
+    end
+
+    if stanza.attr.type == 'result' and sent_iq_cache:get(stanza.attr.id) then
+        sent_iq_cache:set(stanza.attr.id, nil);
+        return true;
+    end
+
+    if stanza.attr.type ~= 'set' then
+        return;
+    end
+
+    local visitors_iq = event.stanza:get_child('visitors', 'jitsi:visitors');
+    if not visitors_iq then
+        return;
+    end
+
+    if origin.type ~= 's2sout' then
+        module:log('warn', 'not from component or s2sin session, ignore! %s %s', origin.type, stanza);
```

Comment:
The code checks for 's2sout' but the log says 's2sin'. Is this intentional?

Can you check the `from` field here? I'm worried that someone could spoof promotion responses by logging into another vnode or the main prosody